File: solutions/python/grains/2/grains.py
'''
function for grains.py
'''
def square(number):
    # when the square value is not in the acceptable range        
    if number <= 0 or number > 64:    
        raise ValueError("square must be between 1 and 64")
    return 2**(number - 1)


def total():
    c = 0
    for i in range(64):
        c += 2**i
    return c

File: solutions/c/grains/2/grains.c
#include "grains.h"

uint64_t square(uint8_t index)
{
    return (index < 1 || 64 < index) ? 0 : 1ul << (index - 1);
}

uint64_t total(void)
{
    uint64_t sum = 0;
    for (int i = 1; i <= 64; i++)
    {
        sum += square(i);
    }
}